src/target/riscv/riscv.c

```diff
@@ -129,6 +129,7 @@ struct scan_field select_idcode = {
 
 bscan_tunnel_type_t bscan_tunnel_type;
 int bscan_tunnel_ir_width; /* if zero, then tunneling is not present/active */
+int bscan_tunnel_ir_id; /* IR ID of the JTAG TAP to access the tunnel. Valid when not 0 */
```

Comment:
```suggestion
static int bscan_tunnel_ir_id = 0x23; /* IR ID of the JTAG TAP to access the tunnel. Valid when not 0 */
```

Reply:
> static int bscan_tunnel_ir_id = 0x23;

I had it to 0 (disabled by default) to ensure backward compatibility (no argument => Ok for some Xilinx devices). 


The following : 
```c
static int bscan_tunnel_ir_id = 0x23;
+ 
uint32_t ir_user4_raw = bscan_tunnel_ir_id != 0 ? bscan_tunnel_ir_id : 0x23 << (target->tap->ir_length - 6);
```
would be problematic, as it would never trigger the current upstream behaviour, right ?